RubrikSecurityCloud Solution initial submission

benmeadowcroft opened the pull request:
Required items, please complete
   
   Change(s):
   - Initial submission of the Rubrik Security Cloud solution for Microsoft Sentinel

   Reason for Change(s):
   - This change enables Rubrik Security Cloud customers to integrate insights from Rubrik’s Data Observability services into Microsoft Sentinel. The insights include identification of anomalous filesystem behavior associated with ransomware and mass deletion, assess the blast radius of a ransomware attack, and sensitive data operators to prioritize and more rapidly investigate potential incidents. The Playbooks provide additional capabilities to operators to recover specific files and scan for Indicators of Compromise across the history of immutable backups to identify safe recovery points

   Testing Completed:
   - Yes, associated custom functions are included in this PR

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

v-sabiraj commented:
@devikamehra / @anki-narravula, please review Playbooks, thanks.

v-sabiraj commented:
Hey @benmeadowcroft, can you please add sample data for Rubrik_Anomaly_Data_CL, Rubrik_Ransomware_Data_CL and Rubrik_ThreatHunt_Data_CL  in [sample data folder](https://github.com/Azure/Azure-Sentinel/tree/master/Sample%20Data/Custom), to test the queries. Please refer other sample data for refernce, thanks.

v-sabiraj commented on Solutions/RubrikSecurityCloud/SolutionMetadata.json:
```diff
@@ -0,0 +1,17 @@
+     "support": {
+         "name": "Rubrik",
+         "email": "support@rubrik.com",
+         "tier": "Partner-supported (Rubrik)",
```
@benmeadowcroft, tier should be just "Partner". Please check and update, thanks.

benmeadowcroft replied:
Thank you, have addressed in Thank you, have resolved in https://github.com/Azure/Azure-Sentinel/pull/6150/commits/f01f79705e6676807acffc5aaecd469db6f54990

v-sabiraj commented:
Please change text from "Azure Sentinel" to "Microsoft Sentinel" in readme files and in description, thanks.

v-sabiraj commented:
Hey @benmeadowcroft, tested function and there is error, please check and update, thanks.
![image](https://user-images.githubusercontent.com/94349919/190392428-2764ba68-5dff-4d77-8aa1-379d3ac7597c.png)

benmeadowcroft commented on Solutions/RubrikSecurityCloud/Data/Solution_RubrikSecurityCloud.json:
```diff
@@ -0,0 +1,23 @@
+{
+    "Name": "RubrikSecurityCloud",
+    "Author": "{Ben Meadowcroft - ben.meadowcroft@rubrik.com}",
```
Thank you, have addressed in https://github.com/Azure/Azure-Sentinel/pull/6150/commits/f01f79705e6676807acffc5aaecd469db6f54990

anki-narravula reviewed (changes requested):
Hi @benmeadowcroft - Please find my feedback below

1. Custom Connector:  Can you please use the standard parameter name like: CustomConnectorName (instead of Custom Apis_Rubrik Custom Connector_name) and please replace at all 3 places you are using this

	- Computation is not allowed for Url fields so  serviceUrl should not have concat, have this derived in a variable and use it

3. Playbooks: in metadata section:
	- prerequisitesDeployTemplateFile - include and provide path for custom connector like "prerequisitesDeployTemplateFile": "../../ThreatXCustomConnector/azuredeploy.json",
	- Entities like "ip", "url" -  "entities": [ "ip", "url" ],
	- tags - like "Enrichment", "Rubrik", "Anomaly" etc - "tags": [ "Enrichment", "ThreatX", "WAF", "Cloud WAF", "Block IP", "Block URL" ]
4. Playbooks: metadata section:
	prerequisites: looks like you are using keyvault and instructions are not given to create KV and what are the secret names need to be created
5. Make sure readme files also capture the requirement of Key Vault
6. Hope we are using V2 version of tool to create solution package. Use version 2.0.0 for first version of the solution

v-sabiraj commented:
> Hey @benmeadowcroft, tested function and there is error, please check and update, thanks. ![image](https://user-images.githubusercontent.com/94349919/190392428-2764ba68-5dff-4d77-8aa1-379d3ac7597c.png)

@benmeadowcroft, please check on this issue. Thanks.

benmeadowcroft commented on Solutions/RubrikSecurityCloud/Data Connectors/RubrikWebhookEvents_API_FunctionApp.json:
```diff
@@ -0,0 +1,171 @@
+{
+    "id": "Rubrik",
```
Addressed by https://github.com/Azure/Azure-Sentinel/pull/6150/commits/1ebcd2bd65048d80bdc0a4e463a98d9e0fce28b1#diff-466207db73fed9899bbb3abde589f1e6551382ee0c9f693c5f9659d5f868d1c6

benmeadowcroft commented:
@v-sabiraj and @anki-narravula please see the fixes addressing your comments in
https://github.com/Azure/Azure-Sentinel/pull/6150/commits/1ebcd2bd65048d80bdc0a4e463a98d9e0fce28b1

Let me know if there are any remaining issues that I can share with the team

v-sabiraj commented:
Hey @benmeadowcroft, thanks for making the suggested changes. There are few corrections.
1. Can you please rename the sample data files as Rubrik_Anomaly_Data_CL, Rubrik_Ransomware_Data_CL and Rubrik_ThreatHunt_Data_CL rather than Rubrik_Anomaly_Event_Data_CL, Rubrik_Ransomware_Event_Data_CL and Rubrik_ThreatHunt_Event_Data_CL. As we need to keep the sample data file names and table name from queries consistent, thanks.

benmeadowcroft commented:
> Hey @benmeadowcroft, thanks for making the suggested changes. There are few corrections.
> 
> 1. Can you please rename the sample data files as Rubrik_Anomaly_Data_CL, Rubrik_Ransomware_Data_CL and Rubrik_ThreatHunt_Data_CL rather than Rubrik_Anomaly_Event_Data_CL, Rubrik_Ransomware_Event_Data_CL and Rubrik_ThreatHunt_Event_Data_CL. As we need to keep the sample data file names and table name from queries consistent, thanks.

@v-sabiraj your requested changes have been made in https://github.com/Azure/Azure-Sentinel/pull/6150/commits/60b2df0963c18597e2c5efd9cb0421dae879ec1f

v-sabiraj commented:
Hey @benmeadowcroft, we need to package the solution with Version 2 of solution package tool, thanks.

benmeadowcroft commented:
> Hey @benmeadowcroft, we need to package the solution with Version 2 of solution package tool, thanks.

@v-sabiraj can you confirm what we need to do here. When I spoke with Eric B (on PM side) recently he mentioned that the packaging may be something Microsoft have to do due to the use of a feature that isn’t supported by the tool yet. I just want to check on this to confirm what needs to happen with this next step and there’s no crossed wires.

v-sabiraj commented:
@benmeadowcroft ok, then in that case can you please remove the package from this PR. So that we can merge this and we will share the package with Eric, thanks.

benmeadowcroft commented:
> @benmeadowcroft ok, then in that case can you please remove the package from this PR. So that we can merge this and we will share the package with Eric, thanks.

I have removed the zip file in https://github.com/Azure/Azure-Sentinel/pull/6150/commits/4539442885586234cd00764292a2c26e579e9208

Is there anything else that needs to be removed or is that sufficent?

benmeadowcroft commented:
> Hey @benmeadowcroft, you need to remove createUiDefinition.json and mainTemplate.json as well. Thanks.

Certainly, removed those files in [52d072b](https://github.com/Azure/Azure-Sentinel/pull/6150/commits/52d072bfebdfade0e6a4c5b10b7c96269771713d)

benmeadowcroft commented:
@v-sabiraj please let me know if any further action is required on my side for this. Thank you

v-sabiraj reviewed (approved):
looks good
